tests/moduleTests/stream.ts

```diff
@@ -883,4 +883,42 @@ export default function tests() {
         assert.equal(s3_n1.name, 'D');
         assert.equal(s3_n1.duration.quarterLength, 4.0);
     });
+
+    test('music21.stream.Score setSubstreamRenderOptions', assert => {
+        const p1 = music21.tinyNotation.TinyNotation('4/4 c4 c c c d d d d e e e e');
+        const p2 = music21.tinyNotation.TinyNotation('4/4 c4 c c c d d d d e e e e');
+        const s = new music21.stream.Score();
+        s.insert(0, p1);
+        s.insert(0, p2);
+
+        // get initial widths
+        s.setSubstreamRenderOptions();
+        const p1_m1 = p1.measures.get(0);
+        const p1_m2 = p1.measures.get(1);
+        const p1_m3 = p1.measures.get(2);
+        assert.notOk(p1_m2.renderOptions.displayClef);
+
+        // force a system break using a margin small enough for clef reiteration to matter
+        const maxWidth = (p1_m1.renderOptions.width + p1_m2.renderOptions.width) - 20;
+        const s_iter = s.recurse(
+            {streamsOnly: true, skipSelf: false}
+        ) as music21.stream.iterator.RecursiveIterator<music21.stream.Stream>;
+        for (const substream of s_iter) {
+            substream.renderOptions.maxSystemWidth = maxWidth;
+            substream.renderOptions.scaleFactor = {x: 1.0, y: 1.0};
+        }
```

Comment:
I think we probably need to test this with other scaleFactors as well, if we want to be sure.